protocol/messenger_test.go

```diff
@@ -2447,3 +2449,38 @@ type testTimeSource struct{}
 func (t *testTimeSource) GetCurrentTime() uint64 {
 	return uint64(time.Now().Unix())
 }
+
+func (s *MessengerSuite) TestSendMessageMention() {
+	// Initialize Alice and Bob's messengers
+	alice, bob := s.m, s.newMessenger()
+	_, err := bob.Start()
+	s.Require().NoError(err)
+	defer bob.Shutdown() // nolint: errcheck
+
+	// Set Bob's profile keypair And set display names for Bob and Alice
+	bobProfileKp := accounts.GetProfileKeypairForTest(true, false, false)
+	bobProfileKp.KeyUID, bobProfileKp.Accounts[0].KeyUID = bob.account.KeyUID, bob.account.KeyUID
+	s.Require().NoError(bob.settings.SaveOrUpdateKeypair(bobProfileKp))
+	s.Require().NoError(bob.SetDisplayName("bobby"))
+	s.Require().NoError(alice.settings.SetDisplayName("Alice"))
```

Comment:
We could remove `SetDisplayName` and `SetNotificationEnabled` functions because they're being used just for testing.

```diff
modified   protocol/messenger_test.go
@@ -24,6 +24,7 @@ import (
 	enstypes "github.com/status-im/status-go/eth-node/types/ens"
 	"github.com/status-im/status-go/images"
 	"github.com/status-im/status-go/multiaccounts/accounts"
+	"github.com/status-im/status-go/multiaccounts/settings"
 	"github.com/status-im/status-go/protocol/common"
 	"github.com/status-im/status-go/protocol/protobuf"
 	"github.com/status-im/status-go/protocol/requests"
@@ -2462,8 +2463,8 @@ func (s *MessengerSuite) TestSendMessageMention() {
 	bobProfileKp.KeyUID, bobProfileKp.Accounts[0].KeyUID = bob.account.KeyUID, bob.account.KeyUID
 	s.Require().NoError(bob.settings.SaveOrUpdateKeypair(bobProfileKp))
 	s.Require().NoError(bob.SetDisplayName("bobby"))
-	s.Require().NoError(alice.settings.SetDisplayName("Alice"))
-	s.Require().NoError(alice.settings.SetNotificationEnabled(true))
+	s.Require().NoError(alice.settings.SaveSettingField(settings.DisplayName, "Alice"))
+	s.Require().NoError(alice.settings.SaveSettingField(settings.NotificationsEnabled, true))
```